csharp/ColumnDescriptor.cs

```diff
@@ -59,9 +64,9 @@ public TReturn Apply<TReturn>(IColumnDescriptorVisitor<TReturn> visitor)
         /// LogicalType is the most nested logical type (e.g. string).
         /// ElementType is the type represented by the column (e.g. string[][][]).
         /// </summary>
-        public (Type physicalType, Type logicalType, Type elementType) GetSystemTypes()
+        public (Type physicalType, Type logicalType, Type elementType) GetSystemTypes(LogicalTypeFactory typeFactory, Type? columnLogicalTypeOverride)
```

Comment:
could we have a defaulted overload or default values on this? I guess most of the time you'll want LogicalTypeFactory.Default and null.

Reply:
Given that it's part of the public api I mean